src/corelibs/U2Core/src/datatype/primers/PrimerStatistics.cpp

```diff
@@ -57,6 +57,23 @@ QString PrimerStatistics::checkPcrPrimersPair(const QByteArray &forward, const Q
     return message;
 }
 
+double PrimerStatistics::getDeltaG(const QByteArray& sequence) {
+    CHECK(validate(sequence), Primer::INVALID_TM);
+
+    double freeEnergy = 0.0;
+    for (int i = 0; i < sequence.size() - 1; i++) {
+        QByteArray curArray;
+        curArray.append(sequence.at(i));
+        curArray.append(sequence.at(i + 1));
```

Comment:
The code is mega inefficient. I see that the QMap<QByteArray, > is an old code, but at least here you can be better by
1) Move QByteArray out of the loop. 
2) Resize it to 2 elements.
3) Re-use it in the loop and set array[0]=sequence[i], array[1]=sequence[i+1]

I know there are other places with the old code like this and I do not ask changing all old places  but only not add a new code like this.

Reply:
OK